### DISCUSSION TITLE
How to query to refetch automatically?

### DISCUSSION THREAD

**Discussion by arslanalidev** [TanStack/query#2534] [Q&A]
I want to stop the refetch of query automatically. Is there any way where we can stop it.

**Comment by TkDodo** [comment]
there are many ways. What do you consider a "refetch", how do you trigger it, and what use-case are you trying to solve please?

**Reply by arslanalidev** [reply]
I want to fetch the data once only when the component did mount occur. After that, it should not fetch the data automatically.

**Reply by TkDodo** [reply]
have a look at the `staleTime` setting and the [important defaults](https://react-query.tanstack.com/guides/important-defaults) section in the docs
